### PR TITLE
Add cache hash support

### PR DESCRIPTION
**Pull request by shyim**
I run multiple jobs for different repositories in one project so i want to manipulate the cache hash

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -71,12 +75,16 @@ export async function restoreCaches(cacheDir: string): Promise<void> {
...
-export async function uploadCaches(cacheDir: string): Promise<void> {
+export async function uploadCaches(
+  cacheDir: string,
+  additionalCacheHash: string
+): Promise<void> {
   try {
     await cache.saveCache(
       [cacheDir],
-      `${process.env['RUNNER_OS']}-qodana-${process.env['GITHUB_REF']}-${process.env['GITHUB_SHA']}`
+      `${process.env['RUNNER_OS']}-qodana-${process.env['GITHUB_REF']}-${additionalCacheHash}`
```
Comment (shyim) [inline src/utils.ts:87]: The previous line didn't make any sense to me as restore, and upload key was not same

Comment (tiulpin) [reply src/utils.ts:87]: Cool idea. Was thinking of this too.
But how about we have `process.env['GITHUB_SHA']`by default (or any other key) and then we use custom keys if they were provided

Comment (shyim) [reply src/utils.ts:87]: You mean setting the default of ` additional-cache-hash` to `${{ github.sha }}`?

Comment (tiulpin) [reply src/utils.ts:87]: Yes, that will do

Comment (tiulpin) [reply src/utils.ts:87]: Don't forget to bump the package.json version to 4.2.0

File: README.md [hunk 2]
```diff
@@ -99,6 +99,7 @@ You can set up [GitHub code scanning](https://docs.github.com/en/code-security/c
 | `profile-path`             | Absolute path to the profile file. Optional.                                                                                       | -                                       |
 | `upload-result`            | Upload Qodana results as an artifact to the job. Optional.                                                                         | `true`                                  |
 | `use-caches`               | Utilize GitHub caches for Qodana runs. Optional.                                                                                   | `true`                                  |
+| `additional-cache-hash`    | Allows customizing the generated cache hash. Optional.                                                                             |                                         |
```
Comment (tiulpin) [inline README.md:102]: maybe name it `custom-cache-key`? Any other shorter names are appreciated 🙂

File: src/utils.ts (hunk 1 above)
Comment (shyim) [reply src/utils.ts:87]: done

**Review by tiulpin — APPROVED** [review]
LGTM!

**Pull request merged**